src/main/org/deidentifier/arx/io/ImportAdapterJDBC.java

```diff
@@ -1,309 +1,419 @@
+
+        super(config);
+        this.config = config;
+
+        /* Preparation work */
+        indexes = getIndexesToImport();
+        dataTypes = getColumnDatatypes();
+
+        try {
+
+            /* Used to keep track of progress */
+            statement = config.getConnection().createStatement();
+
+            String tableName = config.getTable();
+            String schemaName = config.getSchemaName();
+            String dbName = config.getDbName();
+            DatabaseMetaData meta = config.getConnection().getMetaData();
+            String productName = meta.getDatabaseProductName().toLowerCase();
+
+            String afterTableName = "";
+
+            String databaseType = "";
+
+            if (productName.contains("oracle")) {
+                databaseType = "oracle";
+            } else if (productName.contains("db2")) {
+                databaseType = "db2";
+            } else if (productName.startsWith("microsoft sql server")) {
+                databaseType = "sqlserver";
+            } else if (productName.contains("mysql")) {
+                databaseType = "mysql";
+            } else {
+                databaseType = "other";
+            }
+
+            switch (databaseType) {
+                case "mysql":
+                    afterTableName = "`" + schemaName + "`.`" + tableName + "`";
+                    break;
+                case "oracle":
+                    afterTableName = "\"" + schemaName + "\".\"" + tableName + "\"";
+                    break;
+                case "db2":
+                    afterTableName = "\"" + schemaName + "\".\"" + tableName + "\"";
+                    break;
+                case "sqlserver":
+                    afterTableName = "[" + dbName + "].[" + schemaName + "].[" + tableName + "]";
+                    break;
+                default:
+                    afterTableName = tableName;
+                    break;
+            }
+
+
+            statement.execute("SELECT COUNT(*) FROM " + afterTableName);
+            resultSet = statement.getResultSet();
+
+            if (resultSet.next()) {
+
+                totalRows = resultSet.getInt(1);
+
+                if (config.getMaxRows() != 0) {
+                    if (totalRows > config.getMaxRows()) {
+                        totalRows = config.getMaxRows();
+                    }
+                }
+
+                if (totalRows == 0) {
+                    closeResources();
+                    throw new IOException("Table doesn't contain any rows");
+                }
+
+
+            } else {
+                closeResources();
+                throw new IOException("Couldn't determine number of rows");
+            }
+
+            /* Query for actual data */
+            statement = config.getConnection().createStatement();
+
+
+            String querySql = "SELECT * FROM " + afterTableName;
+
+            /* 实现limit逻辑 */
+            if (config.getMaxRows() != 0) {
+
+                switch (databaseType) {
```

Comment:
Do you have a reference showing that this will work for all supported database management systems? If yes, please add as comment